framework/DataObjects/DataSet.py

```diff
@@ -247,6 +247,13 @@ def addVariable(self,varName,values,classify='meta'):
       self._inputs.append(varName)
     elif classify == 'output':
       self._outputs.append(varName)
+      if type(values[0]) == xr.DataArray:
+        indexes = values[0].sizes.keys()
+        for index in indexes:
+          if index in self._pivotParams.keys():
+            self._pivotParams[index].append(varName)
+          else:
+            self._pivotParams[index]=[varName]
```

Comment:
isn't this already implemented? 
I think it is or I am mistaken?

Reply:
If I remember correctly, this is not implemented when using addVariable. Without these changes, the code will crash if we try to reload the DataObject if a new variable is added to the DataObject using addVariable method, since the protect variable ```self._pivotParams``` is not updated, and the index for the new variable can not be founded.